implementations/rust/ockam/ockam_app/src/tcp/outlet/create.rs

```diff
@@ -11,22 +13,40 @@ pub async fn tcp_outlet_create(
     app: AppHandle<Wry>,
     service: String,
     port: String,
-) -> tauri::Result<()> {
-    info!(%service, %port, "Creating an outlet");
-    let state = app.state::<AppState>();
+) -> Result<(), String> {
+    tcp_outlet_create_impl(app, service, port)
+        .await
+        .map_err(|e| {
+            error!("{:?}", e);
+            e.to_string()
+        })?;
+    Ok(())
+}
+
+async fn tcp_outlet_create_impl(
+    app: AppHandle<Wry>,
+    service: String,
+    port: String,
+) -> crate::Result<()> {
+    debug!(%service, %port, "Creating an outlet");
+    let app_state = app.state::<AppState>();
     let tcp_addr = format!("127.0.0.1:{port}")
         .parse()
-        .expect("Invalid IP address");
-    let worker_addr = extract_address_value(&service).expect("Invalid service address");
-    let mut node_manager = state.node_manager.get().write().await;
-    let res = node_manager
-        .create_outlet(&state.context(), tcp_addr, worker_addr, None, true)
-        .await;
-    match res {
-        Err(e) => error!("{:?}", e),
-        Ok(s) => info!("the outlet status is {:?}", s),
+        .into_diagnostic()
+        .wrap_err("Invalid IP address")?;
+    let worker_addr = extract_address_value(&service).wrap_err("Invalid service address")?;
+    let mut node_manager = app_state.node_manager.get().write().await;
+    match node_manager
+        .create_outlet(&app_state.context(), tcp_addr, worker_addr, None, true)
+        .await
+    {
+        Ok(status) => {
+            info!(tcp_addr = status.tcp_addr, "Outlet created");
```

Comment:
We need to start moving the names from `outlet` to `shared_service`. The whole module should be `shared_service` rather than `tcp/outlet` in order to align with the app conceptual model

Reply:
I didn't do that on purpose for the following reason: 
- From the user perspective, they only see services
- Internally, a shared service is a combination of a tcp-outlet running on the local node + a relay running on the project node 

If we start calling "shared services" the code that is handling tcp-outlets specifically, we are goigng to blur the separation between those concepts, which could be confusing. Perhaps we should group tcp-outlet and relay code under a "shared_service" module, so we can have that concept represented properly.

What do you think?

Reply:
> Perhaps we should group tcp-outlet and relay code under a "shared_service" module, so we can have that concept represented properly

Yes, that's what I meant. We should have some top-level modules reflecting the app structure and names

Reply:
Ready!